src/inmanta/protocol/rest/__init__.py

```diff
@@ -219,23 +298,25 @@ async def process(self) -> None:
         for i, arg in enumerate(args):
             arg_type: Optional[type[object]] = self._argspec.annotations.get(arg)
             if arg in self._message:
-                # Argument is parameter in body of path of HTTP request
-                if (
-                    arg_type
-                    and self._properties.operation == "GET"
-                    and typing_inspect.is_generic_type(arg_type)
-                    and issubclass(typing_inspect.get_origin(arg_type), list)
-                    and not isinstance(self._message[arg], list)
-                    and len(typing_inspect.get_args(arg_type, evaluate=True)) == 1
-                    and isinstance(self._message[arg], typing_inspect.get_args(arg_type)[0])
-                ):
-                    # If a GET endpoint has a parameter of type list that is encoded as a URL query parameter and the
-                    # specific request provides a list with one element, urllib doesn't parse it as a list.
-                    # Map it here explicitly to a list.
-                    value = [self._message[arg]]
+                message_arg = self._message[arg]
+
+                if self._properties.operation == "GET":
+                    if self.is_list_or_optional_list(arg_type):
+                        arg_type_not_optional = self.transform_optional_list_to_list(arg_type)
+                        # Update the argument type for optional lists.
+                        arg_type_actual = get_args(arg_type_not_optional)[0]
+                        # Check if the argument type is an enum and get the enum value if applicable.
+                        is_enum = self._is_enum_type(arg_type_actual)
+                        enum_value = self._get_enum_value(arg_type_actual, message_arg) if is_enum else None
+
+                        value = self._handle_generic_list(message_arg, enum_value, is_enum)
+                    else:
+                        value = message_arg
```

Comment:
Why is all the `enum` related logic required here? The conversion from str to enum value is normally done using pydantic in the `validate_arguments()` method below. We explicitly introduced Pydantic in our code base to not have to write this type conversion logic. Could you elaborate?